### ISSUE TITLE
Weight map in crossentropy loss

### ISSUE THREAD

**Issue by Florian-40** [MIC-DKFZ/nnUNet#1274]
Hi Fabian, 

First of all, thanks a lot for this amazing nnUNet. The extent of the tasks it can accomplish is very impressive. I am trying to segment cells in stacks captured by SBF-SEM, but some segmentation are really bad. 

In order to improve inference, I would like to add to crossentropy implementation a weight map as described in the first paper about UNet ([https://arxiv.org/abs/1505.04597](https://arxiv.org/abs/1505.04597)) written by Ronneberger et al. These weights are not the same in a same class. I have written a Python function which return the desired weight map by using a training segmentation. Here you can see what my function can return for one slice of the stack.

![Capture d’écran du 2023-01-11 15-52-01](https://user-images.githubusercontent.com/102385322/211837576-3ef69b26-e634-4ee8-89b9-56db4103816a.png) 

I think that I have to modify the RobustCrossEntropyLoss class to re-implement manually softmax function and my custom crossentropy with weights but I don't know where I have to generate the weight map of each training stack. Probably, where the training labels are loaded after all preprocessing modification ?

Thanks a lot for your help, 

Florian

**Comment by GregorKoehler** [comment]
Hi @Florian-40,
on behalf of all nnUNet maintainers, we want to apologize for leaving this issue stale for so long! Were you successful in creating weight maps for your training? 
In the meantime, nnUNet v2 came out. Are you working in v2 by now?
nnUNet does not support such weight maps, but if you found a good way to integrate them this might be worth a pull request?

**Comment by GregorKoehler** [comment]
Closing this issue for now, as it seems, there's no more need for this. Feel free to re-open if you're still interested in adding this!
